zerver/tests/test_channel_fetch.py

```diff
@@ -763,6 +763,62 @@ def test_gather_partial_subscriptions(self) -> None:
                 self.assertIsNone(sub.get("partial_subscribers"))
                 break
 
+        for sub in subscribed_streams:
+            # fewer than MIN_PARTIAL_SUBSCRIBERS_CHANNEL_SIZE subscribers,
+            # so we get all of them
+            if sub["name"] == "subscribed_more_than_bots_including_idle":
+                self.assertNotIn("partial_subscribers", sub)
+                self.assert_length(sub["subscribers"], 4)
+            if sub["name"] == "subscribed_many_more_than_bots":
+                # the bot, Othello (who is not long_term_idle), and current user
+                self.assert_length(sub["partial_subscribers"], 3)
+                self.assertNotIn("subscribers", sub)
+
+    @override_settings(MIN_PARTIAL_SUBSCRIBERS_CHANNEL_SIZE=5)
+    def test_gather_partial_subscriptions_api(self) -> None:
+        othello = self.example_user("othello")
+        idle_users = [
+            create_user(
```

Comment:
Creating a user is expensive. Do you really need to do it 5 times for this test, or can you just use some of the built-in users?